File: dofun/safe/deviceinfo/utils/FileUtils.java
package com.dofun.safe.deviceinfo.utils;

import android.util.Log;
import java.io.BufferedReader;
import java.io.File;
import java.io.FileReader;
import java.io.IOException;

/* loaded from: classes6.dex */
public class FileUtils {
    public static String readFile(String name) {
        return readFile(new File(name));
    }

    public static String readFile(File file) {
        FileReader fileReader = null;
        BufferedReader reader = null;
        try {
            try {
                StringBuffer sb = new StringBuffer();
                fileReader = new FileReader(file);
                reader = new BufferedReader(fileReader);
                while (true) {
                    String line = reader.readLine();
                    if (line == null) {
                        break;
                    }
                    sb.append(line);
                    sb.append('\n');
                }
                if (sb.length() > 0) {
                    sb.deleteCharAt(sb.length() - 1);
                }
                String stringBuffer = sb.toString();
                try {
                    reader.close();
                } catch (IOException e) {
                    e.printStackTrace();
                }
                try {
                    fileReader.close();
                } catch (IOException e2) {
                    e2.printStackTrace();
                }
                return stringBuffer;
            } catch (Exception e3) {
                Log.d("FileUtils", "readFile error.Just print stack trace.");
                e3.printStackTrace();
                if (reader != null) {
                    try {
                        reader.close();
                    } catch (IOException e4) {
                        e4.printStackTrace();
                    }
                }
                if (fileReader == null) {
                    return Constants.UNKNOWN;
                }
                try {
                    fileReader.close();
                    return Constants.UNKNOWN;
                } catch (IOException e5) {
                    e5.printStackTrace();
                    return Constants.UNKNOWN;
                }
            }
        } catch (Throwable th) {
            if (reader != null) {
                try {
                    reader.close();
                } catch (IOException e6) {
                    e6.printStackTrace();
                }
            }
            if (fileReader != null) {
                try {
                    fileReader.close();
                } catch (IOException e7) {
                    e7.printStackTrace();
                }
            }
            throw th;
        }
    }
}

File: app/src/main/java/com/dofun/safe/deviceinfo/utils/FileUtils.kt
package com.dofun.safe.deviceinfo.utils

import android.util.Log
import java.io.BufferedReader
import java.io.File
import java.io.FileReader
import java.io.IOException

object FileUtils {
    fun readFile(name: String): String {
        return readFile(File(name))
    }

    fun readFile(file: File): String {
        var fileReader: FileReader? = null
        var reader: BufferedReader? = null
        try {
            val sb = StringBuffer()
            fileReader = FileReader(file)
            reader = BufferedReader(fileReader)
            while (true) {
                val line = reader.readLine() ?: break
                sb.append(line)
                sb.append('\n')
            }
            if (sb.isNotEmpty()) {
                sb.deleteCharAt(sb.length - 1)
            }
            val result = sb.toString()
            try {
                reader.close()
            } catch (e: IOException) {
                e.printStackTrace()
            }
            try {
                fileReader.close()
            } catch (e: IOException) {
                e.printStackTrace()
            }
            return result
        } catch (e: Exception) {
            Log.d("FileUtils", "readFile error.Just print stack trace.")
            e.printStackTrace()
            if (reader != null) {
                try {
                    reader.close()
                } catch (closeException: IOException) {
                    closeException.printStackTrace()
                }
            }
            if (fileReader == null) {
                return Constants.UNKNOWN
            }
            return try {
                fileReader.close()
                Constants.UNKNOWN
            } catch (closeException: IOException) {
                closeException.printStackTrace()
                Constants.UNKNOWN
            }
        } catch (throwable: Throwable) {
            if (reader != null) {
                try {
                    reader.close()
                } catch (closeException: IOException) {
                    closeException.printStackTrace()
                }
            }
            if (fileReader != null) {
                try {
                    fileReader.close()
                } catch (closeException: IOException) {
                    closeException.printStackTrace()
                }
            }
            throw throwable
        }
    }
}
